### PR TITLE
Statistics page improvements + fixes

### PR DESCRIPTION
**Pull request by cacraig**
Added charting tool to Statistics page, and fixed bugs with student/educator number reporting.

### CODE REVIEW

**Comment by OpenGlobe** [comment]
@khandelwal can you take a look at this when you have a free moment and merge if it looks ok?  Thank you!

File: ekip/redemption/views.py [hunk 1]
```diff
@@ -43,6 +44,74 @@ def convert_to_date(s):
     """ Convert the date into a useful format. """
     return datetime.strptime(s, '%m/%d/%Y')
 
+def convert_to_db_date(s):
+    """ Convert the date into a useful format. """
```
Comment (khandelwal) [inline ekip/redemption/views.py]: Useful format or database format?

File: ekip/redemption/views.py [hunk 2]
```diff
@@ -43,6 +44,74 @@ def convert_to_date(s):
...
+    date = convert_to_date(s)
+    return date.strftime('%Y-%m-%d')
+
+def get_tickets_by_states(start_date, end_date):
```
Comment (khandelwal) [inline ekip/redemption/views.py:51]: What does this function do? A docstring would  be helpful.

File: ekip/redemption/views.py [hunk 3]
```diff
@@ -43,6 +44,74 @@ def convert_to_date(s):
...
+            tickets_states.append({'state':ticket['recreation_site__state'], 'count':ticket['count']})
+
+    tickets_states = json.dumps(tickets_states)
+    return tickets_states
+
+def get_tickets_by_dates(start_date, end_date):
+
+    # Base query with a date range.
+    ticket_date_query = Ticket.objects \
+                .extra(select={
```
Comment (khandelwal) [inline ekip/redemption/views.py]: Is there not a more Python/ORM way to do this?

File: Procfile [hunk 4]
```diff
@@ -1 +1 @@
-web: cd ekip && python manage.py migrate --settings=config.settings.production --noinput && python manage.py collectstatic --settings=config.settings.production --noinput && waitress-serve --port=$VCAP_APP_PORT config.wsgi:application
+web: cd ekip && python manage.py migrate --settings=config.settings.production --noinput && python manage.py collectstatic --settings=config.settings.production --noinput && waitress-serve --port=$PORT config.wsgi:application
```
Comment (khandelwal) [inline Procfile]: Is this change intentional? Just want to make sure this isn't creeping in as part of statistics page improvements.

**Comment by khandelwal** [comment]
There don't seem to be corresponding unit tests for any of the Python changes in here. It'd be great to have those.

**Comment by cacraig** [comment]
Thanks, @khandelwal I made the updates according to your feedback. I also broke out that Procfile change to another branch.  https://github.com/18F/ekip-api/pull/245 .
